src/library_noderawfs.js

```diff
@@ -93,15 +93,12 @@ mergeInto(LibraryManager.library, {
         throw new FS.ErrnoError(ERRNO_CODES.ENOTDIR);
       }
       var newMode = NODEFS.getMode(pathTruncated);
-      var fd = suggestFD != null ? suggestFD : FS.nextfd(nfd);
       var node = { id: st.ino, mode: newMode, node_ops: NODERAWFS, path: path }
-      var stream = FS.createStream({ nfd: nfd, position: 0, path: path, flags: flags, node: node, seekable: true }, fd, fd);
-      FS.streams[fd] = stream;
-      return stream;
+      return FS.createStream({ nfd: nfd, position: 0, path: path, flags: flags, node: node, seekable: true }, nfd);
```

Comment:
Looking at this diff, I think this should be:
```suggestion
      return FS.createStream({ nfd: nfd, position: 0, path: path, flags: flags, node: node, seekable: true }, -1);
```

i.e. before this change, `suggestFD` was always `null`, meaning it would pick an empty slot in the `FS.streams` array as a result of the `FS.nextfd()` call.

(The reason for this comment is due to a [suspicious test failure on wasm-vips](https://github.com/kleisauke/wasm-vips/actions/runs/5143202220/jobs/9267769739#step:7:295) after updating to 3.1.40 from 3.1.39)

Reply:
Oh I see.. `nfd` means `native fd` ?  And I guess noderawfs uses its own FD -> NFD mapping?   Sounds reasonable to change this.   Do you have a test case that we could use?